### ISSUE TITLE
Runtime crash on update with create subquery

### ISSUE THREAD

**Issue by IlyaSemenov** [romeerez/orchid-orm#139]
After #105, it's now possible to update with a subquery result (thanks!).

However, the typings are too permissive. A subquery of `create` is accepted but leads to runtime crash:

```ts
// No type errors
await db.user.find(1).update({
  name: db.user.get('name').create({ name: 'Bob' })
})
```

```
UPDATE "user" SET "name" = (INSERT INTO "user"("name") VALUES ($1) RETURNING "user"."name") WHERE "user"."id" = $2 ['Bob', 1]
Error: syntax error at or near "INTO"
```

Similarly, it crashes on `update` subquery:

```ts
// No type errors, but runtime crash
await db.user.find(1).update({
  name: db.user.find(1).get('name').update({ name: 'Bob' }),
})
```

I think it should be possible to disallow `SetQueryReturns` in the typings?

---

Ideally, `.update()` could detect such sub-queries and handle them with `with xxx as`:

```sql
with _insert1 as (INSERT INTO "user" (name) VALUES ('Bob') RETURNING name)
update "user" set name = (select name from _insert1) where id=1
```

but this is probably quite a challenge and I'd totally understand if you say this is out of scope of the ORM.

**Comment by romeerez** [comment]
Ready! Here are [the docs for it](https://orchid-orm.netlify.app/guide/create-update-delete.html#sub-queries).

Thank you for suggesting `WITH` - did it exactly in this way. Such query should work fine:

```ts
await db.user.find(1).update({
  name: db.user.get('name').create({ name: 'Bob' })
})
```

`update` and `delete` as well are acceptable here.

And for the case when trying to do this for related tables, it won't work because `WITH` cannot reference the main query table, I added TS and runtime errors for such case:

```ts
await db.user.find(1).update({
  // TS error
  name: (q) => q.relatedTable.get('name').create({ name: 'Bob' }),
})
```

**Comment by IlyaSemenov** [comment]
Awesome!! And super powerful. I imagine I would have waited for things like that getting implemented in competitors for months if not years.
